packages/@aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-listener.ts

```diff
@@ -26,7 +26,7 @@ export interface BaseNetworkListenerProps {
    *
    * @default - TLS if certificates are provided. TCP otherwise.
    */
-  readonly protocol?: Protocol;
+  readonly protocol?: NetworkProtocol;
```

Comment:
Sorry, but we can't have this. I don't think this is a good enough reason to break API compatibility.

I would be more in favor of adding `TCP_UDP` to the existing `Protocol` enum, noting in the docstring that it is only for NLBs, and adding runtime validation to make sure it isn't being used for ALBs.